### ISSUE TITLE
[Proposal] Apply infer option to field member or apply it as new option.

### ISSUE THREAD

**Issue by RevensofT** [dotnet/vblang#515]
I'm not sure anyone suggest this topic but I think infer field member should be apply, any reason it apply to local variant should be use with initialized field member.

Befor.
```vb
Public Shared DInvoke As Func(Of Int32, Int32, Int32, Int32, Int32, Int32, Int32) = Friendly.DInvoke

Friend Shared Copy_block_byte As Action(Of UIntPtr, UIntPtr, uint) = "la.1.0.2 cpyb".compile(Of Action(Of UIntPtr, UIntPtr, uint)).fin
```

After
```vb
Public Shared DInvoke = Friendly.DInvoke

Friend Shared Copy_block_byte = "la.1.0.2 cpyb".compile(Of Action(Of UIntPtr, UIntPtr, uint)).fin
```

**Comment by RevensofT** [comment]
But VS dev start to integrate AI and Machine learning into VS, doesn't it ? why not use them to solve problem over 10 years ago ?

**Comment by aarondglover** [comment]
The article says its all... It's not impossible but would require major effort for minimal benefits.

**Comment by RevensofT** [comment]
Many thing start from small benefit, example in VS 16, dev develop AI intellisense to help us write a  code we frequency write, doesn't it require major effort for minimal benefits too ?

**Comment by Happypig375** [comment]
Compared to rewriting the whole compiler, adding AI to Intellisense is a small effort.

**Comment by tverweij** [comment]
@Happypig375 : Yes it is

**Comment by tverweij** [comment]
AI intellisense - I am curious what that will do.
Till now, all Microsoft "help" with what I would mean only makes life harder for me.

Like this site I have issue 515 open, so if I am typing 51 in the address bar, it will assume that I mean 515 - and open the wrong page.

So I hope we can switch it off.

**Comment by Happypig375** [comment]
@tverweij Anything that you type in the address bar is autocompleted by your browser based on your site visit frequency, not this site's fault lol

**Comment by tverweij** [comment]
Nope, it's the fault of the "help" I get from the browser - that is what I meant.
Never asked for autocomplete - it's only harassing me,.

**Comment by tverweij** [comment]
And I am afraid that this AI will do something like that; did three times this, so now your are going to do that again - NOT.

**Comment by Happypig375** [comment]
Except the suggestions provided by AI IntelliSense (IntelliCode) can be ignored easily. https://visualstudio.microsoft.com/services/intellicode/

**Comment by tverweij** [comment]
Perfect
